src/ckb-daemon/usb_linux.c

```diff
@@ -67,7 +68,11 @@ static char kbsyspath[DEV_MAX][FILENAME_MAX];
 ///
 int os_usbsend(usbdevice* kb, const uchar* out_msg, int is_recv, const char* file, int line) {
     int res;
-    if ((kb->fwversion >= 0x120 || IS_V2_OVERRIDE(kb)) && !is_recv){
+    // If we need to read a response, lock the mutex so that os_usbrecv() waits for the main thread.
+    if(is_recv)
+        pthread_mutex_lock(&kb->interruptmutex);
```

Comment:
Consider checking for errors.

Reply:
... here, and in other pthread_* functions, no point in commenting under every call.